streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesSessionClientV2.scala

```diff
@@ -133,4 +165,115 @@ object KubernetesSessionClientV2 extends KubernetesClientV2Trait with Logger {
     }
   }
 
+  private def genFlinkConfig(deployRequest: DeployRequest): Configuration = {
+    val flinkConfig =
+      extractConfiguration(deployRequest.flinkVersion.flinkHome, deployRequest.properties)
+    flinkConfig
+      .safeSet(KubernetesConfigOptions.NAMESPACE, deployRequest.k8sDeployParam.kubernetesNamespace)
+      .safeSet(
+        KubernetesConfigOptions.KUBERNETES_SERVICE_ACCOUNT,
+        deployRequest.k8sDeployParam.serviceAccount)
+      .safeSet(
+        KubernetesConfigOptions.REST_SERVICE_EXPOSED_TYPE,
+        ServiceExposedType.valueOf(deployRequest.k8sDeployParam.flinkRestExposedType.getName))
+      .safeSet(KubernetesConfigOptions.CLUSTER_ID, deployRequest.clusterId)
+      .safeSet(KubernetesConfigOptions.CONTAINER_IMAGE, deployRequest.k8sDeployParam.flinkImage)
+      .safeSet(
+        KubernetesConfigOptions.KUBE_CONFIG_FILE,
+        getDefaultKubernetesConf(deployRequest.k8sDeployParam.kubeConf))
+      .safeSet(DeploymentOptionsInternal.CONF_DIR, s"${deployRequest.flinkVersion.flinkHome}/conf")
+  }
+
+  private def genFlinkDeployDef(
+      deployRequest: DeployRequest,
+      originFlinkConfig: Configuration): Either[FailureMessage, FlinkDeploymentDef] = {
+
+    val flinkConfObj = originFlinkConfig.clone()
+    val flinkConfMap = originFlinkConfig.toMap.asScala.toMap
+
+    val namespace = deployRequest.k8sDeployParam.kubernetesNamespace
+
+    val name = deployRequest.k8sDeployParam.clusterId
+
+    val imagePullPolicy = flinkConfObj
+      .getOption(KubernetesConfigOptions.CONTAINER_IMAGE_PULL_POLICY)
+      .map(_.toString)
+
+    val image = deployRequest.k8sDeployParam.flinkImage
+    val serviceAccount = deployRequest.k8sDeployParam.serviceAccount
```

Comment:
`serviceAccount` need to fallback with `FlinkDeploymentDef.DEFAULT_SERVICE_ACCOUNT`